packages/client/hmi-client/src/utils/bitmask.ts

```diff
@@ -0,0 +1,76 @@
+enum OperatorInteractionStatus {
+	Found = 0x0001,
+	Hover = 0x0010,
+	Focus = 0x0100,
+	Drag = 0x1000
+}
+
+function addState(status: number, state: OperatorInteractionStatus) {
+	// eslint-disable-next-line no-bitwise
+	status |= state;
+	return status;
+}
+
+function removeState(status: number, state: OperatorInteractionStatus) {
+	// eslint-disable-next-line no-bitwise
+	status &= ~state;
+	return status;
+}
+
+function isStateActive(status: number, state: OperatorInteractionStatus) {
+	// eslint-disable-next-line no-bitwise
+	return status & state;
+}
+
+// Hover
+export function addHover(status: number) {
+	return addState(status, OperatorInteractionStatus.Hover);
+}
+
+export function removeHover(status: number) {
+	return removeState(status, OperatorInteractionStatus.Hover);
+}
+
+export function isHover(status: number) {
+	return isStateActive(status, OperatorInteractionStatus.Hover);
+}
+
+// Found
+export function addFound(status: number) {
+	return addState(status, OperatorInteractionStatus.Found);
+}
+
+export function removeFound(status: number) {
+	return removeState(status, OperatorInteractionStatus.Found);
+}
+
+export function isFound(status: number) {
+	return isStateActive(status, OperatorInteractionStatus.Found);
+}
```

Comment:
I wonder if writing a class might clean things up a bit. I don't wanna over engineer this yet as this the first pass.

Reply:
if you want this to be less verbose you can use `const` instead of function:
```typescript
export const isFound = (status:number) => isStateActive(status, OperatorInteractionStatus.Found);
```